### ISSUE TITLE
Solidus REST API Headless

### ISSUE THREAD

**Issue by CharlyJazz** [solidusio/solidus#3076]
Is it possible to use Solidus REST API to create SPA, PWA, or simply consult an endpoint to filter products?

I refer to the Front End as such, not using the API to create an Admin panel

**Comment by kennyadsl** [comment]
@CharlyJazz yes, you can include only engine that you need. Instead of:

```
gem 'solidus'
```

You could do:

```
gem 'solidus_core'
gem 'solidus_api'
gem 'solidus_backend'
gem 'solidus_sample'
```

and implement your own frontend.

> I can not find a way to filter a product asynchronously using XMLHTTPREQUEST.

Not sure to get what you mean. Can you please explain better?

**Comment by CharlyJazz** [comment]
I'm trying to filter, or do anything like

Obtain products or taxonomies using the REST API ... etc.

But I can only use the REST API if I have a token from a user Admin and that does not make sense for the Front End. I need to do AJAX because I want to make a modern web with React (Single Page APpl)...

**Comment by kennyadsl** [comment]
Yes you are right, there's no built in way to use the API without an API key. This can be solved in several ways, see. #2808 and the resulting extension https://github.com/skukx/solidus_jwt for an example of implementaion using JWT.

**Comment by CharlyJazz** [comment]
@ericsaupe Thats gold. We need that in the documentation. The people want create PWA, and modern front end architectures.

**Comment by ericsaupe** [comment]
Definitely something that can be better documented. I'll see if I can get a PR open with some better documentation for this gotcha. We did a single page checkout using the `solidus_api` and ran into this. 

Thanks for the question!

**Comment by CharlyJazz** [comment]
Thank for u support bro! @ericsaupe  :tada: :tada:

**Comment by kennyadsl** [comment]
We recently merged #3133 and I think we can close this one since our API is fully documented. Thanks for your question @CharlyJazz!
